diff --git a/immich_ml/models/ocr/recognition.py b/immich_ml/models/ocr/recognition.py
index 6408e4818fde8..94f40c928596f 100644
--- a/immich_ml/models/ocr/recognition.py
+++ b/immich_ml/models/ocr/recognition.py
@@ -64,6 +64,7 @@ def _load(self) -> ModelSession:
                 rec_batch_num=max_batch_size if max_batch_size else 6,
                 rec_img_shape=(3, 48, 320),
                 lang_type=self.language,
+                model_root_dir=self.cache_dir,
             )
         )
         return session
