--- a/onnxruntime/core/platform/posix/env.cc
+++ b/onnxruntime/core/platform/posix/env.cc
@@ -201,7 +201,7 @@ class PosixThread : public EnvThread {
  private:
   static void* ThreadMain(void* param) {
     std::unique_ptr<Param> p(static_cast<Param*>(param));
     ORT_TRY {
-#if !defined(__APPLE__) && !defined(__ANDROID__) && !defined(__wasm__) && !defined(_AIX)
+#if !defined(__APPLE__) && !defined(__ANDROID__) && !defined(__wasm__) && !defined(_AIX) && !defined(__FreeBSD__)
       if (p->affinity.has_value() && !p->affinity->empty()) {
         cpu_set_t cpuset;
         CPU_ZERO(&cpuset);

--- a/tools/python/util/platform_helpers.py
+++ b/tools/python/util/platform_helpers.py
@@ -12,4 +12,4 @@ def is_macOS():  # noqa: N802


 def is_linux():
-    return sys.platform.startswith("linux")
+    return sys.platform.startswith("linux") or sys.platform.startswith("freebsd")
